client/src/linkManager.js

```diff
@@ -88,6 +88,37 @@ export class linkManager extends LuigiClientBase {
     helpers.sendPostMessageToLuigiCore(navigationOpenMsg);
   }
 
+  /**
+   * Offers an alternative way of navigating with intents. This involves specifying a semanticSlug and an object containing
+   * parameters.
+   * This method internally generates a URL of the form `#?intent=<semantic object>-<action>?<param_name>=<param_value>` through the given
+   * input arguments. This then follows a call to the original `linkManager.navigate(...)` function.
+   * Consequently, the following calls shall have the exact same effect:
+   * - linkManager().navigateToIntent('Sales-settings', {project: 'pr2', user: 'john'})
+   * - linkManager().navigate('/#?intent=Sales-settings?project=pr2&user=john')
+   * @param {string} semanticSlug concatenation of semantic object and action connected with a dash (-), i.e.: `<semanticObject>-<action>`
+   * @param {Object} params an object representing all the parameters passed, i.e.: `{param1: '1', param2: 2, param3: 'value3'}`
+   * @example
+   * LuigiClient.linkManager().navigateToIntent('Sales-settings', {project: 'pr2', user: 'john'})
+   */
+  navigateToIntent(semanticSlug, params) {
+    let newPath = '#?intent=';
+    newPath += semanticSlug;
```

Comment:
Do we have a potential XSS attack possibility here?

Reply:
This is currently not possible to happen I think. I think this was already prevented when we added intent-based navigation the first time around. 
[In this line](https://github.com/SAP/luigi/blob/master/core/src/utilities/helpers/routing-helpers.js#L346) we check for the semantic-object with an alphanumeric regexp. 
While the parameters we also sanitize them when we build the node. Like [here](https://github.com/SAP/luigi/blob/master/core/src/utilities/helpers/routing-helpers.js#L81) and [here](https://github.com/SAP/luigi/blob/master/core/src/navigation/services/navigation.js#L165) . Basically just tried it out and they get sanitized properly.